src/fetchtastic/download/base.py

```
@@ -117,6 +127,223 @@ def download(self, url: str, target_path: Pathish) -> bool:
+
+        Returns:
+            bool: True if download succeeded, False otherwise.
+        """
+        delay = retry_delay
+
+        for attempt in range(max_retries + 1):
+            try:
+                result = await self.async_download(url, target_path, progress_callback)
+                if result:
+                    return True
+            except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)